book/scheduling-and-threading.md

```diff
@@ -195,34 +206,53 @@ def set_timeout(func, sec):
     t.start()
 ```
 
-Next, add a dirty bit called `needs_pipeline_update` (plus `display_scheduled`
-to avoid double-running `set_timeout` unnecessarily) to `Tab`, which means
-"rendering needs to happen, and has been scheduled, but hasn't happened yet".
- 
-Also, rename `render` to `run_rendering_pipeline`, and add a new
-`run_animation_frame` method that runs the pipeline and calls the other
-rendering pipeline stages on `Browser` via a new `raster_and_draw`
-method (which we'll implement shortly).
+Next, add three dirty bits to `Tab`:
+
+* `needs_pipeline_update`, indicating that
+the pipeline needs to be re-run.
+* `display_scheduled`, indicating that `set_timeout` was already called (this
+avoids double-running `set_timeout` unnecessarily).
+* `run_pipeline_now`, indicating that the event loop should
+run the pipeline.
+
+Add methods to set the dirty bits and call `set_timeout` as
+needed.[^race-condition]
+
+[^race-condition]: This code has a race condition because it's setting variables
+on one thread from another without a thread lock. Let's ignore that for now and
+fix it later in the chapter.
```

Comment:
Potentially we should move discussion of locks way up to where we first introduce threads (setTimeout) so that we don't have to continuously apologize for it throughout the book.

Reply:
ok, good point